spec.html

```diff
@@ -24069,6 +24069,34 @@ <h1>Object.defineProperty ( _O_, _P_, _Attributes_ )</h1>
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-Object.fromEntries">
+        <h1>Object.fromEntries ( iterable )</h1>
+        <p>When the `fromEntries` method is called with argument _iterable_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _obj_ be ObjectCreate(%ObjectPrototype%).
+          1. Assert: _obj_ is an extensible ordinary object with no own properties.
+          1. If _iterable_ is *undefined* or *null*, return _obj_.
```

Comment:
This differs from the behavior we originally had, which was to throw. This behavior contradicts [this point](https://github.com/tc39/proposal-object-from-entries/blob/master/DETAILS.md#non-optionality-of-argument) in the DETAILS document.

This was a normative change included as part of [this commit](https://github.com/tc39/proposal-object-from-entries/commit/3b1c261276289a66f78cfb148b729479e1fdf540#diff-3540caefa502006d8a33cb1385720803), I'm guessing by accident. (That commit does what the Map constructor does, but we explicitly want to do something different.) I think the fix is to change this to `If _iterable_ is *undefined* or *null*, throw a TypeError exception` and move it to the first step.

Reply:
Note that this same confusion was brought up and addressed [here](
https://github.com/tc39/proposal-object-from-entries/pull/20#discussion_r204927185)

Reply:
It does not have the same semantics. `GetIterator` throws if passed undefined or null (via [GetIterator](https://tc39.github.io/ecma262/#sec-getiterator) -> [GetMethod](https://tc39.github.io/ecma262/#sec-getmethod) -> [GetV](https://tc39.github.io/ecma262/#sec-getv) -> [ToObject](https://tc39.github.io/ecma262/#sec-toobject)).

This is about the case where `GetIterator` is passed `undefined`, not an object whose `Symbol.iterator` property is `undefined`.

Reply:
aha, thank you. Can you file this on the proposal repo itself, and I'll address it there before updating here?

Reply:
[Filed](https://github.com/tc39/proposal-object-from-entries/issues/22)

Reply:
https://github.com/tc39/proposal-object-from-entries/pull/23